File: LeetCode/[M] - 1493. Longest Subarray of 1's After Deleting One Element/solution.go
package main

func longestSubarray(nums []int) int {
	mx, curr, l, zIdx := 0, 0, 0, 0
	canDelete := true
	for r := 0; r < len(nums); r++ {
		if nums[r] == 0 && canDelete {
			zIdx = r
			canDelete = false
		} else if nums[r] == 0 && !canDelete {
			mx = max(mx, curr-1)
			l = zIdx + 1
			zIdx = r
		}
		curr = r - l + 1
	}
	return max(mx, curr-1)
}

File: LeetCode/[M] - 1493. Longest Subarray of 1's After Deleting One Element/solution.py
class Solution:
    def longestSubarray(self, nums: list[int]) -> int:
        mx, curr, l, zdx = 0, 0, 0, -1
        can_delete = True
        for r in range(len(nums)):
            if nums[r] == 0 and can_delete:
                zdx = r
                can_delete = False
            elif nums[r] == 0 and not can_delete:
                mx = max(mx, curr - 1)
                l = zdx + 1
                zdx = r
            curr = r - l + 1
        return max(mx, curr - 1)
